js/jquery.validation.js

```diff
@@ -143,32 +181,38 @@
                 var ajax_method;
                 if ( method === 'POST' || method === 'GET' ) {
                     ajax_method = method;
+
                 } else {
                     // Emulate HTTP method
-                   ajax_method = 'POST';
-                    var found = false;
-                    $.each(data, function(index, item) {
-                        if ( item.name === '_method' ) {
-                            item.value = method;
-                            found = true;
-                            return;
-                        }
-                    });
+                    ajax_method = 'POST';
+
+                    if ( isFormData ) {
+                        data.set('_method',method);
```

Comment:
Yes, if you're on a browser that supports `FormData` but not `FormData.prototype.set` and calling the `validate` method directly, with the PUT method, and passing in a FormData object instead of a serialized data object, an error will be thrown and the method will fail.

The "submit" event handler already copes with that by the second approach to constructing form data.

I don't think that (as yet) we have any instances of a developer calling the `validate` method directly with `FormData` at all.

I don't think there's anything we can do in that situation, except throw an error. We could perhaps document more clearly that calling the `validate` method with a `FormData` object cannot support methods other than POST and GET unless the browser supports FormData.prototype.set, and that any developer who wants to use FormData in this manner needs to check compatibility first.